packages/reown_core/lib/pairing/pairing.dart

```diff
@@ -513,6 +522,83 @@ class Pairing implements IPairing {
     }
   }
 
+  ///
+  /// Sign 2.5
+  /// Substitutes wc_sessionPropose sendRequest()
+  /// during signEngine.connect()
+  ///
+  @override
+  Future<dynamic> sendProposeSessionRequest(
+    String topic,
+    Map<String, dynamic> params, {
+    int? id,
+    EncodeOptions? encodeOptions,
+  }) async {
+    final proposeSessionPayload = JsonRpcUtils.formatJsonRpcRequest(
+      MethodConstants.WC_SESSION_PROPOSE,
+      params,
+      id: id,
+    );
+    final requestId = proposeSessionPayload['id'] as int;
+
+    final proposeSessionMessage = await core.crypto.encode(
+      topic,
+      proposeSessionPayload,
+      options: encodeOptions,
+    );
+
+    if (proposeSessionMessage == null) {
+      return;
+    }
+
+    // print('adding payload to pending requests: $requestId');
+    final resp = PendingRequestResponse(
+      completer: Completer(),
+      method: MethodConstants.WC_SESSION_PROPOSE,
+    );
+    resp.completer.future.catchError((err) {
+      // Catch the error so that it won't throw an uncaught error
+    });
+    pendingRequests[requestId] = resp;
+
+    final payload = {
+      'pairingTopic': topic,
+      'sessionProposal': proposeSessionMessage,
+    };
+
+    final options = PublishOptions(
+      correlationId: requestId,
+      ttl: null, // ttl is not required in this case, it's assigned relay-side
+      tag: null, // tag is not required in this case, it's assigned relay-side
```

Comment:
Is there a concept of default values in Flutter? Maybe we could assign null to ttl and tag if that's the case